src/generators/dart/DartFileGenerator.ts

```diff
@@ -11,14 +11,15 @@ export class DartFileGenerator extends DartGenerator implements AbstractFileGene
    * @param input
    * @param outputDirectory where you want the models generated to
    * @param options
+   * @param ensureFilesWritten verify that the files is completely written before returning, this can happen if the file system is swamped with write requests. 
    */
-  public async generateToFiles(input: Record<string, unknown> | InputMetaModel, outputDirectory: string, options: DartRenderCompleteModelOptions): Promise<OutputModel[]> {
+  public async generateToFiles(input: Record<string, unknown> | InputMetaModel, outputDirectory: string, options: DartRenderCompleteModelOptions, ensureFilesWritten = false): Promise<OutputModel[]> {
```

Comment:
Hmm, maybe that last argument can be a part of options? In this place and another ones? I have only problem with that, rest code is good 👌🏼

Reply:
Yea it's a good point, I actually almost commit that change here 🙃 (have it in my git stash) However I decided to do it in another PR that unifies all option behaviors 🙂

Reply:
Ok, I will accept but remember about this next PR :D

Reply:
Will do, gonna create an issue first to propose it 🙂